application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -94,54 +99,111 @@ public BotCore(JDA jda, Database database, Config config) {
             .toList();
 
         // User Interactors (e.g. slash commands)
-        nameToInteractor = features.stream()
+        interactors = features.stream()
             .filter(UserInteractor.class::isInstance)
             .map(UserInteractor.class::cast)
-            .collect(Collectors.toMap(UserInteractor::getName, Function.identity()));
+            .filter(validateInteractor())
+            .toList();
+
+        nameToInteractor = interactors.stream()
+            .collect(Collectors.toMap(UserInteractorPrefix::getPrefixFromInstance,
```

Comment:
it reads as if the key would be just the prefix, but instead its the "full name including a prefix". either rename the method or add some ` + name` to it.
Alternatively, change the maps key from `String` to something like a `record UserInteractorId(String prefix, String name)` or similar. if we keep the type high level and not a raw `String`, its easier to not mess up

Reply:
Would be amazing if someone else also had an opinion on this, I get you, but I just dont feel like it I guess.
I'll do a quick check on variable names and documentation so it's clear. I'll stick with that if that's fine?

Reply:
yeah, its fine